src/main/java/org/datadog/jenkins/plugins/datadog/DatadogBuildListener.java

```diff
@@ -690,23 +581,6 @@ public void setTargetMetricURL(String targetMetricURL) {
             this.targetMetricURL = targetMetricURL;
         }
 
-        /**
-         * @return - A {@link StatsDClient} lease for this registered {@link RunListener}
-         */
-        public StatsDClient leaseStatDClient() {
-            try {
-                if (statsDClient == null) {
-                    statsDClient = new NonBlockingStatsDClient("jenkins.job", daemonHost.split(":")[0],
-                            Integer.parseInt(daemonHost.split(":")[1]));
-                } else {
-                    logger.warning("StatsDClient is already set");
-                }
-            } catch (Exception e) {
-                logger.severe(String.format("Error while configuring StatsDClient. Exception: %s", e.toString()));
-            }
-            return statsDClient;
-        }
-
         /**
          * @return - A {@link DatadogClient} lease for this registered {@link RunListener}
          */
```

Comment:
This also keeps logging `logger.warning("datadogClient is already set");` because we have `DatadogClient client = getDescriptor().leaseDatadogClient();` in 2 methods (`OnStarted` and `OnCompleted`).